.github/workflows/cut-rc.yml

```diff
@@ -0,0 +1,547 @@
+              continue
+            fi
+
+
+            # Find commits with the specific commit message after the fork point
+            # to not cherry-pick twice the same commit.
+            ALREADY_PICKED_CHERRIES=$(git log \
+              --grep "^(cherry picked from commit $COMMIT)\$" --format='%H' \
+              HEAD...$FORK_POINT)
+
+            if [ -z "$ALREADY_PICKED_CHERRIES" ]; then
+              if git cherry-pick -x "$COMMIT"; then
+                echo "Cherry-picked $COMMIT from PR $PR" >> $GITHUB_STEP_SUMMARY
+
+                [ "${{github.event.inputs.dry-run}}" != "true" ] && \
+                gh pr comment $PR --body "Merge commit has been cherry-picked to branch \`${{needs.variables.outputs.branch}}\`."
+
+                EMPTY=false
+              else
+                git cherry-pick --abort
+
+                [ "${{github.event.inputs.dry-run}}" != "true" ] && \
+                gh pr comment $PR --body "Please merge the changes to branch \`${{needs.variables.outputs.branch}}\`."
+
+                cat << EOF >> $GITHUB_STEP_SUMMARY
+          * [PR $PR]($URL) by **${AUTHOR}** (_${TITLE}_) could not be cherry-picked. Merge commit: \`$COMMIT\`.
+          EOF
+                echo "- <$URL|PR $PR> by *$AUTHOR* — $TITLE" \
+                >> "$SLACK_MESSAGE_FILE"
+              fi
+            else
+              echo "Already picked cherries for commit $COMMIT:"
+              echo "$ALREADY_PICKED_CHERRIES"
+              echo "..."
+            fi
+          done
+
+          git push
+          if [ -n "$EMPTY" ]; then
+            echo "Cherry-picked commits have been pushed to the release branch." >> $GITHUB_STEP_SUMMARY
+          fi
+
+          # Replace % with %25 and join lines with %0A:
+          FAILED="$(sed ':a; N; $!ba; s/%/%25/g; s/\n/%0A/g' $SLACK_MESSAGE_FILE)"
+          rm -f "$SLACK_MESSAGE_FILE"
+
+          if [ -n "$FAILED" ]; then
+            echo "::set-output name=bad-cherries::$FAILED"
+            cat << EOF >> $GITHUB_STEP_SUMMARY
+
+          As some of the PRs could not be cherry-picked, please help the authors to merge their changes to the release branch via opening PRs.
+          The commands may look like the following:
+
+              git pull
+              git switch "${{needs.variables.outputs.branch}}"
+              git switch --create <AUTHOR>/merge-pr-<PR NUMBER>-"${{needs.variables.outputs.release}}"
+              git cherry-pick -x <MERGE COMMIT SHA>
+              # Proceed to resolve merge conflicts. Once done:
+              git push --set-upstream origin $(git branch --show-current)
+
+              # Create PR to the release branch:
+              gh create pr --base "${{needs.variables.outputs.branch}}" --fill \\
+                --milestone "${{needs.variables.outputs.milestone}}"
+
+          EOF
+            exit 1
+          fi
+      - name: Post to Slack
+        if: always() && steps.cherry-pick.outputs.bad-cherries != ''
```

Comment:
`if: always()`?
What does it mean 🤯 ?

Reply:
github actions magic, this is required to not skip the step after previous step failure. Could probably be changed to `failure()`.